### PR TITLE
DEPS: Unpin pydata-sphinx-theme

### LINKED ISSUE #48988: CI: pydata sphinx theme 0.11 broke doc build
See https://github.com/pandas-dev/pandas/actions/runs/3203777448/jobs/5234277313

On this note: I think it would be a good idea to test against main again. This would enable us to catch bugs sooner

### LINKED ISSUE #53167: CI/BLD: Doc builds are failing
Starting in the last hour or so all doc builds on PRs are failing with the following error:

```
Theme error:
An error happened in rendering the page development/community.
Reason: UndefinedError("'logo' is undefined")
```

### PR DESCRIPTION
**Pull request by mroeschke**
- [ ] closes #48988 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

closes #53167

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -237,14 +237,14 @@
...
+    "footer_start": ["pandas_footer", "sphinx-version"],
     "github_url": "https://github.com/pandas-dev/pandas",
     "twitter_url": "https://twitter.com/pandas_dev",
-    "google_analytics_id": "UA-27880019-2",
+    "analytics": {"google_analytics_id": "UA-27880019-2"},
     "logo": {"image_dark": "https://pandas.pydata.org/static/img/pandas_white.svg"},
     "navbar_end": ["version-switcher", "theme-switcher", "navbar-icon-links"],
     "switcher": {
-        "json_url": "/versions.json",
+        "json_url": "versions.json",
```
Comment (datapythonista) [inline doc/source/conf.py:247]: Seems like the dropdown is broken with this change: https://pandas.pydata.org/preview/53029/docs/

What was the problem that made you change this?

Comment (mroeschke) [reply doc/source/conf.py:247]: Seems like in 0.13 this is the code that reads the json_url:

https://github.com/pydata/pydata-sphinx-theme/blob/f15ecfed59a2a5096c05496a3d817fef4ef9a0af/src/pydata_sphinx_theme/__init__.py#L105

With the `/version.json` for example in https://github.com/pandas-dev/pandas/actions/runs/4863923942/jobs/8672280782, I would get

```
2023-05-02T17:22:30.6605921Z WARNING: The version switcher "/versions.json" file cannot be read due to the following error:
2023-05-02T17:22:30.6606819Z FileNotFoundError(2, 'No such file or directory')
```

Comment (datapythonista) [reply doc/source/conf.py:247]: Ah, I see. Not great, but then I think we need this:

```suggestion
        "json_url": "https://pandas.pydata.org/versions.json",
```

Comment (mroeschke) [reply doc/source/conf.py]: Great this worked thanks!

**Comment by datapythonista** [comment]
JFYI you can click on the link of the previous `/preview` comments, it's always the same link, and the command doesn't generate the preview, only the comment with the link for convenience. :) The preview is generated by the docs CI job directly. You don't even need the command if you don't mind typing the url with the PR number.

**Comment by mroeschke** [comment]
Ah okay sorry I thought the command also had to re-point the link to the latest web artifact.

Thanks for the prior suggestion, the version dropdown seems to work now

**Comment by rhshadrach** [comment]
When you select 1.5, I'm seeing the docs for 1.5.3:

> Date: Nov 22, 2022 Version: 1.5.2

**Comment by mroeschke** [comment]
Yeah I think this is the same behavior that currently happens: https://pandas.pydata.org/pandas-docs/version/1.5/index.html

Since this fixed the doc build issue, going to merge, but happy to address any followups

**Pull request merged**